### PR TITLE
Support fixture kwargs for generator_fixture

### PR DESCRIPTION
**Pull request by ayalash**

### CODE REVIEW

**Comment by vmalloc** [comment]
We need to add tests for the new usage

**Comment by ayalash** [comment]
@vmalloc An unittest was added + entry to changelog.
Please review.

File: slash/core/fixtures/utils.py [hunk 1]
```diff
@@ -87,12 +87,15 @@ def generator_fixture(func):
...
+    if func is None:
+        return functools.partial(generator_fixture, **kw)
+
     @parametrize('param', list(func()))
     def new_func(param):
         return param
 
-    new_func.__name__ = func.__name__
-    return _ensure_fixture_info(func=new_func)
+    kw.setdefault('name', func.__name__)
```
Comment (vmalloc) [inline slash/core/fixtures/utils.py]: Is this necessary? How did it work until now if this wasn't here?

File: doc/changelog.rst [hunk 2]
```diff
@@ -1,6 +1,7 @@
 Changelog
 =========
 
+* :feature:`-` Supports fixture kwargs for ``generator_fixture``
```
Comment (vmalloc) [inline doc/changelog.rst]: `s/Supports/Support/`
`s/kwargs/keyword arguments/`

File: tests/test_generator_fixture.py [hunk 3]
```diff
@@ -1,4 +1,36 @@
...
+        def test_something(a_generator, second_generator):
+            assert a_generator == 5
+            assert second_generator == 10
+
+        session.fixture_store.add_fixture(my_generator)
+        session.fixture_store.add_fixture(second_generator)
+        session.fixture_store.resolve()
+        return session, test_something
+
+    session, test = generate_session_and_test()
```
Comment (vmalloc) [inline tests/test_generator_fixture.py]: You can skip the function part - just declare them directly inside the test...

**Pull request merged**